Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -969,12 +969,7 @@ bool CGPathIsEmpty(CGPathRef path) {
         return true;
     }
 
-    UINT32 count;
-
-    RETURN_FALSE_IF_FAILED(path->ClosePath());
-
-    RETURN_FALSE_IF_FAILED(path->GetPathGeometry()->GetFigureCount(&count));
-    return count == 0;
+    return !path->IsGeometryStarted();
```

Comment:
I've had this in for a while used in other locations. It's just more simple than querying the amount of figures or segments.